apex-squid/src/test/java/org/fundacionjala/enforce/sonarqube/apex/parser/grammar/ApexGrammarConstructorDeclarationTest.java

```diff
@@ -0,0 +1,80 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.fundacionjala.enforce.sonarqube.apex.parser.grammar;
+
+import org.fundacionjala.enforce.sonarqube.apex.parser.ApexRuleTest;
+import org.junit.Before;
+import org.junit.Test;
+import static org.fundacionjala.enforce.sonarqube.apex.api.grammar.ApexGrammarRuleKey.CONSTRUCTOR_DECLARATION;
+import static org.sonar.sslr.tests.Assertions.assertThat;
+
+public class ApexGrammarConstructorDeclarationTest extends ApexRuleTest {
+
+    @Before
+    public void init() {
+        setRootRule(CONSTRUCTOR_DECLARATION);
+    }
+
+    @Test
+    public void testValidConstructorsDeclarations() {
+        assertThat(parser)
+                .matches("something () {"
+                        + "this(blockstatement);"
+                        + "}")
+                .matches("something (integer something) {"
+                        + "super(blockstatement);"
+                        + "}")
+                .matches("myClass () {"
+                        + "}")
+                .matches("myClass (other otherClass) { "
+                        + "super();"
+                        + "}")
+                .matches("myConstructor (integer parameter){}")
+                .matches("transient (integer parameter) {super(parameter);}")
+                .matches("after (Class classParameter) {"
+                        + "this();"
+                        + "}")
+                .matches("data (ClassType parameter) {}")
+                .matches("group (integer parameter) {super(parameter);}");
+
+    }
+
+    @Test
+    public void testInvalidConstructorDeclarations() {
+        assertThat(parser)
+                .notMatches("myClass)( {"
+                        + "this(blockstatement);"
+                        + "}")
+                .notMatches("public classDeclaration(int parameter) {"
+                        + "super(blockstatement);"
+                        + "}")
+                .notMatches("myClass() {"
+                        + "{")
+                .notMatches("myClass(lowercaselasstype otherClass) { "
+                        + "super()"
```

Comment:
Yeah that case is contemplated in the positive tests (above, in line 50). The negative case was when it's missing the semicolon, should I remove it?